metrics/integration_test.go

```diff
@@ -101,6 +102,46 @@ var _ = Describe("Metrics Integration", Ordered, func() {
 			Expect(testutil.CollectAndCompare(SnapshotCreatedToPipelineRunStartedSeconds, strings.NewReader(readerData))).To(Succeed())
 		})
 	})
+
+	Context("When RegisterPipelineRunWithEphemeralEnvStarted is called", func() {
+		// As we need to share metrics within the Context, we need to use "per Context" '(Before|After)All'
+		BeforeAll(func() {
+			// Mocking metrics to be able to resent data with each tests. Otherwise, we would have to take previous tests into account.
+			//
+			// 'Help' can't be overridden due to 'https://github.com/prometheus/client_golang/blob/83d56b1144a0c2eb10d399e7abbae3333bebc463/prometheus/registry.go#L314'
+			SnapshotCreatedToPipelineRunWithEphemeralEnvStartedSeconds = prometheus.NewHistogram(
+				prometheus.HistogramOpts{
+					Name:    "snapshot_created_to_pipelinerun_started_seconds",
```

Comment:
I'm very puzzled with those tests, metric is called `snapshot_created_to_pipelinerun__with_ephemeral_env_started_seconds` in code, here we have another name.

I may need explanation how this test works, like for 8y old.

How `RegisterPipelineRunWithEphemeralEnvStarted` is mocked to use this new variable?
`SnapshotPipelineRunStartedSecondsHeader` is used in this test, don't we need separate header?

Reply:
First of all, there's no reason to take that tone.

Second, I've updated the things you mentioned here as well as a few similar references.  I don't see why the header needs to be changed aside from clarity, especially given that SnapshotPipelineRunStarted will soon become SnapshotPipelineRunWithoutEphemeralEnvStarted.  But it's been changed nonetheless.